core/core.go

```diff
@@ -436,6 +443,23 @@ func (t *TextileNode) WaitForRoom() {
 	}
 }
 
+func (t *TextileNode) GatewayPort() (int, error) {
+	// Get config and set proxy address to raw gateway address plus one thousand,
+	// so a gateway on 8182 means the proxy will run on 9182
+	cfg, err := t.Context.GetConfig()
+	if err != nil {
+		return -1, fmt.Errorf("ServeHTTPGatewayProxy: GetConfig() failed: %s", err)
```

Comment:
error format string not relevant anymore.. maybe just "get gateway port failed: %s"?

Reply:
also, log.Errorf (and the one below)

Reply:
right, moved log above return, and then return 'raw' error